src/Utility/Vec3.h

```diff
@@ -31,6 +31,7 @@ class Vec3
   Real& x; //!< Reference to X-component
   Real& y; //!< Reference to Y-component
   Real& z; //!< Reference to Z-component
+  const Real* params[3] = {};
```

Comment:
Don't see why we need three pointers. Should be sufficient with just `Real* params;` here which then points to an array with parameter values. Or alternatively (maybe also prefarable), consider a Vec3 (or Vec4 maybe( sub-class which have the u, v and w parameters as additional members.

Reply:
as i said this was just quickly tossed together, and in the fe class it was three vars so i just made three pointers so i could point directly (using the ability of them being null to flag searching required). i take it you do not think the approach is too invasive then?